pyBumpHunter/BumpHunter.py

```diff
@@ -286,415 +336,433 @@ def Reset(self):
-            if(self.weights is None):
+            if self.weights is None:
                 bkg_hist = bkg
             else:
                 bkg_hist = bkg * self.weights
             data_hist = data
             Hbins = self.bins
-        
+
         # Generate all the pseudo-data histograms
-        if(do_pseudo):
-            pseudo_hist = np.random.poisson(lam=np.tile(bkg_hist,(self.Npe,1)).transpose(),size=(bkg_hist.size,self.Npe))
-        
+        if do_pseudo:
+            pseudo_hist = np.random.poisson(
+                lam=np.tile(bkg_hist, (self.Npe, 1)).transpose(),
+                size=(bkg_hist.size, self.Npe),
+            )
+
         # Set width_max if it is given as None
         if self.width_max is None:
             self.width_max = data_hist.size // 2
-        
+
         # Initialize all results containenrs
-        if(do_pseudo):
-            self.min_Pval_ar = np.empty(self.Npe+1)
-            self.min_loc_ar = np.empty(self.Npe+1,dtype=int)
-            self.min_width_ar = np.empty(self.Npe+1,dtype=int)
-            self.res_ar = np.empty(self.Npe+1,dtype=object)
+        if do_pseudo:
+            self.min_Pval_ar = np.empty(self.Npe + 1)
+            self.min_loc_ar = np.empty(self.Npe + 1, dtype=int)
+            self.min_width_ar = np.empty(self.Npe + 1, dtype=int)
+            self.res_ar = np.empty(self.Npe + 1, dtype=object)
         else:
-            if(self.res_ar==[]):
+            if self.res_ar == []:
                 self.min_Pval_ar = np.empty(1)
-                self.min_loc_ar = np.empty(1,dtype=int)
-                self.min_width_ar = np.empty(1,dtype=int)
-                self.res_ar = np.empty(1,dtype=object)
-        
+                self.min_loc_ar = np.empty(1, dtype=int)
+                self.min_width_ar = np.empty(1, dtype=int)
+                self.res_ar = np.empty(1, dtype=object)
+
         # Auto-adjust the value of width_max and do an array of all width
-        w_ar = np.arange(self.width_min,self.width_max+1,self.width_step)
+        w_ar = np.arange(self.width_min, self.width_max + 1, self.width_step)
         width_max = w_ar[-1]
-        print('{} values of width will be tested'.format(w_ar.size))
-        
+        print(f"{w_ar.size} values of width will be tested")
+
         # Compute the p-value for data and all pseudo-experiments
         # We must check if we should do it in multiple threads
-        print('SCAN')
-        if(do_pseudo):
-            if(self.Nworker>1):
+        print("SCAN")
+        if do_pseudo:
+            if self.Nworker > 1:
                 with thd.ThreadPoolExecutor(max_workers=self.Nworker) as exe:
-                    for th in range(self.Npe+1):
-                        if(th==0):
-                            exe.submit(self.__scan_hist,data_hist,bkg_hist,w_ar,th)
+                    for th in range(self.Npe + 1):
+                        if th == 0:
+                            exe.submit(self._scan_hist, data_hist, bkg_hist, w_ar, th)
```

Comment:
What does that help? Is there an improvement in speed? I don't think that threads help here. Rather, (if even) processes to circumvent the GIL? Or use something like [ray](https://github.com/ray-project/ray)